openmdao/core/indepvarcomp.py

```diff
@@ -69,31 +70,18 @@ def __init__(self, name=None, val=1.0, **kwargs):
                 raise ValueError("IndepVarComp init: '%s' is not supported "
                                  "in IndepVarComp." % illegal)
 
-    def _post_configure(self):
+    def _configure_check(self):
         """
-        Do any remaining setup that had to wait until after final user configuration.
+        Do any error checking on i/o configuration
         """
-        # set static mode to False because we are doing things that would normally be done in setup
-        self._static_mode = False
-
-        self._var_rel_names = {'input': [], 'output': []}
-        self._var_rel2meta = {}
-
-        for (name, val, kwargs) in self._indep + self._indep_external:
-            super(IndepVarComp, self).add_output(name, val, **kwargs)
-
-        for (name, val, kwargs) in self._indep_external_discrete:
-            super(IndepVarComp, self).add_discrete_output(name, val, **kwargs)
-
+       
         if len(self._indep) + len(self._indep_external) + len(self._indep_external_discrete) == 0:
```

Comment:
I don't think we need all these _indep, _indep_external, or _indep_external_discrete anymore.  If there isn't already a way in the parent class to keep track of the number of outputs (discrete and nondiscrete), then replace all three of these with a single counter or even a boolean that is set to True if `add_output` or `_add_discrete_output` is called.

Reply:
Agreed.  Looks like those just aren't needed anymore at all.  It could just check the length of self._static_var_rel_names['output'] and self._var_rel_names['output'] or the length of self._static_var_rel2meta and self._var_rel2meta.